numpy/ma/core.pyi

```diff
@@ -1154,7 +1154,10 @@ def sort(
     *,
     stable: Literal[False] | None = False,
 ) -> NDArray[Any]: ...
-def compressed(x): ...
+@overload
+def compressed(x: _ArrayLike[_ScalarT_co]) -> ndarray[tuple[int], dtype[_ScalarT_co]]: ...
+@overload
+def compressed(x: ArrayLike) -> ndarray[tuple[int], dtype[Any]]: ...
```

Comment:
oh sorry for not realizing this before, but since #28669 the `dtype[Any]` can be written as `dtype`:

```suggestion
def compressed(x: ArrayLike) -> ndarray[tuple[int], dtype]: ...
```